### PR TITLE
Fix some compilation warnings (and never have them again)

### PR DESCRIPTION
**Pull request by vkucera**
Fix some compilation warnings (some of which were genuine bugs) and add them to the list of compilation errors so that they are caught in CI tests.

I did my best to figure out what the correct code should look like, but I would like to ask the authors of the respective code pieces to cross-check:

- [x] @victorvalenciatorres PWGDQ/Tasks/dqCorrelation.cxx
- [x] @hkoley PWGLF/Tasks/Resonances/lambda1520analysis.cxx
- [x] @altsybee PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx
- [x] @EmilGorm PWGCF/GenericFramework/Core/FlowPtContainer.cxx
- [x] @EmilGorm PWGCF/GenericFramework/Core/BootstrapProfile.cxx
- [x] @shreyasiacharya PWGDQ/Core/HistogramsLibrary.cxx
- [x] @peressounko PWGEM/Tasks/phosNbar.cxx

### CODE REVIEW

File: dependencies/O2PhysicsCompileFlags.cmake [hunk 1]
```diff
@@ -11,7 +11,14 @@
 
 include_guard()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-error -Werror=unused-variable")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-error \
+-Werror=aggressive-loop-optimizations \
+-Werror=deprecated-enum-float-conversion \
+-Werror=maybe-uninitialized \
```
Comment (aalkin) [inline dependencies/O2PhysicsCompileFlags.cmake:17]: This is a GCC-specific option, as well as `-Werror=aggressive-loop-optimizations`, you should add clang's alternatives based on the platform.

**Review by peressounko — APPROVED** [review]

File: PWGLF/Tasks/Resonances/lambda1520analysis.cxx [hunk 2]
```diff
@@ -404,7 +404,7 @@ struct lambda1520analysis {
         }
       }
     }
-    return false;
+    return isTrk1Selected;
```
Comment (njacazio) [inline PWGLF/Tasks/Resonances/lambda1520analysis.cxx:407]: @hkoley this looks like a fixed bug, isn't it?

Comment (hkoley) [reply PWGLF/Tasks/Resonances/lambda1520analysis.cxx:407]: Yes, it should return boolean values for `isTrk1Selected` and `isTrk2Selected`. Thanks for fixing the bug.

**Review by njacazio — DISMISSED** [review]

**Comment by vkucera** [comment]
Ready for merging. (macOS build failure unrelated)

**Review by victorvalenciatorres — APPROVED** [review]

**Review by aalkin — DISMISSED** [review]

**Review by mpuccio — DISMISSED** [review]

**Review by mfasDa — APPROVED** [review]

**Review by feisenhu — APPROVED** [review]

**Comment by jgrosseo** [comment]
Common is fine for me.
@victor-gonzalez I haven't approved because it would also approve CF which is on your side I think

**Review by victor-gonzalez — DISMISSED** [review]
Strange that the compiler does not complain in L.130 of PWGCF/GenericFramework/Core/BootstrapProfile.cxx

**Comment by vkucera** [comment]
> Strange that the compiler does not complain in L.130 of PWGCF/GenericFramework/Core/BootstrapProfile.cxx

That line does not cast `this`.

**Review by aalkin — APPROVED** [review]

**Review by feisenhu — APPROVED** [review]

**Review by mfasDa — DISMISSED** [review]

**Review by njacazio — APPROVED** [review]

**Review by ddobrigk — APPROVED** [review]

**Review by ktf — APPROVED** [review]

**Review by mpuccio — APPROVED** [review]

**Comment by vkucera** [comment]
@ktf @ddobrigk can we force the merge please? The remaining approval concerns 2 unused variables in PWGUD and @pbuehler is now on vacations. I guess he won't mind.

**Review by pbuehler — APPROVED** [review]

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**